### DISCUSSION TITLE
Do I need to put single-atom geometry to training set if I use E0s?

### DISCUSSION THREAD

**Discussion by turbosonics** [ACEsuit/mace#518] [Q&A] [answered]
I get E0s for target atoms using VASP's single point energy calculation. But do I need to include those individual atom geometry (single atom in 15x15x15 angstrom^3 box) in the training set? 

If yes, then is this should be just a single frame (or a single configuration) for each atom type? And what should I need to do for these single-atom configurations in training set xyz file? Do I need to label it?

**Comment by gabor1** [comment, marked as answer]
You can put your isolated atoms in the training file, as separate frames, and they will get picked up by the training script. You can also omit them from the training file, and specify the E0s on the command line.

**Comment by bernstei** [comment]
I think you have to label them with `config_type=IsolatedAtom`, or something like that.

**Reply by bernstei** [reply]
From the README

```
It is usually preferred to add the isolated atoms to the training set, rather than reading in their energies through the command line like in the example above. To label them in the training set, set config_type=IsolatedAtom in their info fields.
```

**Reply by turbosonics** [reply]
Thanks. But I think a bit more details on this one would be really helpful. 
1. Does this single atom need some frame numbers, like 100 images, (or number of configurations) just like other NVT/NPT AIMD data? Or just a single image? I think the single frame would be the answer but just in case I'm asking. 
2. What data is required for the information line (line 2) of xyz file of these single atom images? For example, I'm printing out stress and training E, F, and S. But I wonder if those single atom frames need to have stress tensor in its information line because of format or any other reasons.  
3. Do I need to label all other AIMD geometries differently, if I label single atom geometry as "IsolatedAtom"? 
4. Once I label single atom geometry (and all other AIMD geometries), then do I need to do something to "--config_type_weights" in the input?

**Reply by gabor1** [reply]
1. Not sure what you mean by "frame number". 
2. the single atom frame just needs to be a valid extended XYZ file, so what you put in the second line just needs to be compatible with that. so you put in the lattice and the properties. force is analytically zero, so you don't need to give it (but can if you like), stress would be extremely small (if your unit cell is big enough), again you can omit it or give it, doesn't matter. 
3. there is no interaction with other geometries . obviously don't label non-single-atom frames as IsolatedAtom. 
4. you don't need to specify config_type weights if you don't want to. the E0s will be enforced analytically, so there is no role for any weights in this case.

**Reply by turbosonics** [reply]
> 1. Not sure what you mean by "frame number".

It is just number of xyz images or xyz configurations. To build a training set, I usually bring 4000~5000 xyz images from NVT or NPT AIMD for a content. I was wondering do I need to bring at least hundred number of images for isolated atom to prevent the isolated atom data omitted among thousands of images of AIMD data, but I guess it wouldn't be necessary if we use different config card. 

> 2. the single atom frame just needs to be a valid extended XYZ file, so what you put in the second line just needs to be compatible with that. so you put in the lattice and the properties. force is analytically zero, so you don't need to give it (but can if you like), stress would be extremely small (if your unit cell is big enough), again you can omit it or give it, doesn't matter.

Yeah, I already made force 0, but I was wondering do I need to bring stress or not. I will just make all stress 0.
 
> 3. there is no interaction with other geometries . obviously don't label non-single-atom frames as IsolatedAtom.
> 4. you don't need to specify config_type weights if you don't want to. the E0s will be enforced analytically, so there is no role for any weights in this case.

Thank you! 
But slightly different question, does various weight numbers for different config type recommended for MACE training? So far, I just used {"Default":1.0} but I wonder if this feature may be able to distinguish and give different priority or importance among various AIMD data in the training set. For example, give 100 to NPT with high temperature to make it more important or make MACE prioritize, while give 1 to 300K NVT to make it less important or make MACE less prioritize it...

**Reply by gabor1** [reply]
there is only one possible config for an isoalted atom, it doesn't make sense to repeat it. in any case, as I said the value is enforced analytically, not by the loss function. 

you can indeed use the config weights to place more or less importance to various different config types. I recommend only doing it if you notice that certain kinds of configurations are much less accurately fitted than you'd like (and for forces, remember that it's the relative force accuracy that matters, not the absolute one)

**Reply by turbosonics** [reply]
Thank you so much!
